File: VB/DXExample.Module.Web/WebReportsController.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp

Namespace DXExample.Module.Web
	Public Class WebReportsController
		Inherits ViewController
		Public Sub New()
			TargetViewType = ViewType.DetailView
			TargetObjectType = GetType(MyReportData)
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			AddHandler (CType(View.CurrentObject, MyReportData)).ReportCreated, AddressOf WinReportsController_ReportCreated
		End Sub
		Private Sub WinReportsController_ReportCreated(ByVal sender As Object, ByVal e As ReportCreatedEventArgs)
			' Place your code here
		End Sub
		Protected Overrides Sub OnDeactivated()
			MyBase.OnDeactivated()
			RemoveHandler (CType(View.CurrentObject, MyReportData)).ReportCreated, AddressOf WinReportsController_ReportCreated
		End Sub
	End Class
End Namespace

File: CS/DXExample.Module.Web/WebReportsController.cs
using System;

using DevExpress.ExpressApp;

namespace DXExample.Module.Web {
    public class WebReportsController : ViewController {
        public WebReportsController() {
            TargetViewType = ViewType.DetailView;
            TargetObjectType = typeof(MyReportData);
        }
        protected override void OnActivated() {
            base.OnActivated();
            ((MyReportData)View.CurrentObject).ReportCreated += new EventHandler<ReportCreatedEventArgs>(WinReportsController_ReportCreated);
        }
        void WinReportsController_ReportCreated(object sender, ReportCreatedEventArgs e) {
            // Place your code here
        }
        protected override void OnDeactivating() {
            base.OnDeactivating();
            ((MyReportData)View.CurrentObject).ReportCreated -= new EventHandler<ReportCreatedEventArgs>(WinReportsController_ReportCreated);
        }
    }
}
